File: Basics_with_JavaScript/1314 Nested Loops/More Exersises/02. Letters Combinations.js
function vladi(array) {
  let startNum = array[0];
  let endNum = array[1];
  let badLetter = array[2];
  let counter = 0;
  let combinationLine = "";

  let startNumDigit = startNum.charCodeAt(0);
  let endNumDigit = endNum.charCodeAt(0);
  let badLetterDigit = badLetter.charCodeAt(0);

  for (let first = startNumDigit; first <= endNumDigit; first++) {
    let firstString = String.fromCharCode(first);
    if (first === badLetterDigit) {
      continue;
    }

    for (let second = startNumDigit; second <= endNumDigit; second++) {
      let secondString = String.fromCharCode(second);
      if (second === badLetterDigit) {
        continue;
      }
      for (let third = startNumDigit; third <= endNumDigit; third++) {
        let thirdString = String.fromCharCode(third);
        if (third === badLetterDigit) {
          continue;
        }
        //counter to be moved after all successful rules
        counter++;
        combinationLine += firstString + secondString + thirdString + " ";

        // end of inner loop
      }

      // end of middle loop
    }

    // end of outer loop
  }
  combinationLine = combinationLine + counter;
  console.log(combinationLine);
}

vladi(["a", "c", "b"]);
vladi(["f", "k", "h"]);
vladi(["a", "c", "z"]);

File: Basics_with_Python/PythonBasicCourse/6_Nested_Loops/14_Nested_Loops___Exercise/Nested_Loops___More_Exercises/02._Letters_Combinations.py
start_str = input()
end_str = input()
skip_str = input()
counter = 0
start_num = ord(start_str)
end_num = ord(end_str)
skip_num = ord(skip_str)

for first in range(start_num, end_num + 1):
    if first == skip_num:
        continue
    for second in range(start_num, end_num + 1):
        if second == skip_num:
            continue
        for third in range(start_num, end_num + 1):
            if third == skip_num:
                continue
            print(chr(first)+chr(second)+chr(third), end=" ")
            counter += 1
print(counter)
